agent/script/drain/concrete_script.go

```diff
@@ -96,12 +108,30 @@ func (s ConcreteScript) runOnce(params ScriptParams) (int, error) {
 	command.Args = append(command.Args, jobChange, hashChange)
 	command.Args = append(command.Args, updatedPkgs...)
 
-	stdout, _, _, err := s.runner.RunComplexCommand(command)
+	process, err := s.runner.RunComplexCommandAsync(command)
 	if err != nil {
 		return 0, bosherr.WrapError(err, "Running drain script")
 	}
 
-	value, err := strconv.Atoi(strings.TrimSpace(stdout))
+	var result boshsys.Result
+
+	// Can only wait once on a process but cancelling can happen multiple times
+	for processExitedCh := process.Wait(); processExitedCh != nil; {
+		select {
+		case result = <-processExitedCh:
+			processExitedCh = nil
+		case <-s.cancelCh:
+			// Ignore possible TerminateNicely error since we cannot return it
+			process.TerminateNicely(10 * time.Second)
+			return 0, bosherr.Error("Script was cancelled by user request")
```

Comment:
i dont think we need to return here since process will be terminated and it will return some exit code.